### ISSUE TITLE
Example code in manuscript not easily reproducible

### ISSUE THREAD

**Issue by annakrystalli** [Infectious-Disease-Modeling-Hubs/hubEnsembles#64]
There are a number of instances, most author facing but more importantly, also a couple of reader facing instances where the code to access data is not reproducible when running the code as is interactively.

For example, the following line of code in the  Case study: Weekly incident flu hospitalizations section

```r
flu_forecasts_raw <- readr::read_rds("data/flu_forecasts_raw.rds")
```
This only works if you render the vignette or are working inside of a clone of the `hubEnsembles` package AND have set the working directory to the manuscript directory.

All these are not expectations we can assume a reader of the paper that wants to play around with the code will be aware of.

### Potential solutions

- the easiest solution would be to wrap the paths in function here which always resolves relative to the root of the repository, ege.:
  ```r
  lu_forecasts_raw <- readr::read_rds(here::here("manuscript", "data", "flu_forecasts_raw.rds"))
  ```
  This will work when the Rmd is knit but also whenever someone is running the code interactively
- Separate the manuscript into it's own a research compendium. I personally don't see the need for these materials to be in the hubEnsembles package repo and would suggest moving it to it's own repository that interested readers can download and interact with. Thsi would not only get rid of having to download the package source to interact with manuscript code/data and awkward paths, it also means we could manage manuscript dependencies separately (and more formally). I did have to install a bunch of packages to run the vignette which, wasn't a big deal but felt a bit haphazard.

**Comment by elray1** [comment]
I like Anna's last proposed solution here in many ways.  My one question about it is whether that would mean the document was no longer available as a vignette.  Is that an important consideration for us?

**Comment by nickreich** [comment]
All things being equal, I think it would be better to keep this as a vignette that is part of the package. It would be great to have it hosted as part of the pkgdown website, for example, and it then lives with the package and serves double duty as really showing people how to use it. But I understand the concerns about data size, data packaging, and wanting to keep the collect() workflow in the manuscript.

- what would the size be of the .rda data objects needed for the vignette?
- we could finalize the manuscript and have one static version that lives as a vignette here on the repo that uses the .rda files and another closely related version that we submit to a journal that keeps the `collect()` functionality and that we update as we go through peer review?

**Comment by elray1** [comment]
It looks like the total size of rds files in [this folder](https://github.com/Infectious-Disease-Modeling-Hubs/hubEnsembles/tree/software-manuscript/manuscript/data) is 8.4MB. Those are related to Li's analysis for flu.

I assume rda files with examples could be pretty small.

I kind of like the general idea of potentially maintaining two closely-related versions of the vignette, despite the obvious drawbacks of maintaining two versions of a very similar document.

**Comment by nickreich** [comment]
I haven't looked at the results/application part of the vignette/manuscript yet, so I'm going to wait to read that to make more suggestions/comments until I do.

**Comment by annakrystalli** [comment]
One thing we could also consider for example data is separating them out into a data package, a bit like the [Long Term Ecological Research program (LTER) Network's](https://lternet.edu/) [`lterdatasampler`](https://lter.github.io/lterdatasampler/) 📦 or Allison Horst's [`palmerpenguins`](https://allisonhorst.github.io/palmerpenguins/) 📦.

We could mix rda data as well as example hubs in the `inst`  folder that are accessed via `system.file()`. 

Actually I think this is really the way to go and surprised it only just occured me given I really rate the above mentioned packages! 

I'll add a note to the example data discussion regarding this too.

**Comment by eahowerton** [comment]
Following discussion with @elray1, I have added the `example-complex-forecast-hub` data into the `data` folder for now. I think it could be useful to keep this data with the package (in particular, I have in mind users who are less familiar with the broader hubverse framework/resources). Or, when we have a better solution, we can update accordingly. Seems a broader discussion is on the horizon. 
NOTE: I wasn't sure how long-term this solution would be, so I have not updated the package version. For now, you'll need to rebuild before knitting.

**Comment by elray1** [comment]
Summing up some misc. things that came up in a discussion with Li just now, which don't get to answers but at least attempt to clarify questions:

- There are two "examples" in the manuscript that pull from data: simple examples of hub data for a variety of data types, and Li's thorough analysis of flu forecasts.
- For each of those examples, there are two decisions to make:
    1. Should that example be included in a package vignette?  (it's given that both will be included in the manuscript)
        1. Our answer for the first one, simple examples, is "yes"
        2. If we say also "yes" to Li's fuller flu analysis, the vignette could be the same as the manuscript.  Li noted that we had previously answered "no" to this question.  But we could undo that decision.
    2. Should the data supporting that analysis live inside the hubEnsembles repo?
        1. For the first one, Anna has proposed creating a separate R package housing hubverse example data.  That sounds fine to me, but maybe one thing to discuss/coordinate on across hubverse packages is whether we want precisely the same example data and setup for those data.
        2. For the second one, if we answer "yes" to item i.b., I wonder if one option could be to move all the "original source" data and analysis code out to a separate repo, and include just whatever minimal summaries of flu results are needed to create plots and tables in the manuscript in a subfolder of `inst` in this repo?

**Comment by nickreich** [comment]
I'm still getting through reading the two "in action" sections of the vignette/manuscript, so I don't have a clear answer on the last comment yet. 

That said, I am piling on the questions without answers: I have had the feeling that in the manuscript currently we are loading one very large example file (25K+ rows) and then doing a lot of repeated subsetting for different code chunks in ways that are not always entirely clear. For some of the operations we only subset a bit but then show a snapshot of and describe the top of the file that is visible in the output. This doesn't seem quite right to me, as someone who ran that code would have additional output in their session that is not explained in the vignette. So this suggests to me that either:

1. we need multiple different example data files for hubEnsembles. E.g., one that is just a simple example of a single quantile forecast (for explaining the data format and plotting), one single example of a pmf forecast (for explaining the data format), and then another df that contains outputs from multiple models for showing off ensemble functionality. I think I'd lean towards this, and 
2. we should load the single large example file and then explicitly subset it in the manuscript depending on the task/use case. 

I am leaning towards doing 1, and setting up some additional scripts the way @eahowerton did to create several other smaller example data objects that are part of the hubEnsembles package.

**Comment by lshandross** [comment]
I've just gone through and reviewed all the data used within the manuscript and I have a few thoughts:
1. I agree with Nick about using multiple different example data files for the package, each for different types of examples
2. We don't really need all of the current data files for the case study section, as a subset would suffice based on what we actually are plotting or displaying in tables. There are several options that I see for reducing the amount of data involved in the case study portion that we're actually storing:
   i. Only keep the original, raw forecasts directly pulled from flusight for a single date (which the user could then play around with using the ensemble functions); the raw target data; and the scores for all ensembles and the baseline. This amounts to less than 2 mb. However, this means the user doesn't have access to all of the ensemble forecasts, which includes the ones for california that are plotted (I'm leaning towards this option)
  ii. Further subset the above data to be only the original raw forecasts and target data for California, and the scores to be exactly what's plotted/displayed. This will be the least data, but be less accessible for the user to play around with
  iii. Keep all of the current data files but subset the ensemble forecasts to only be for California, which is what's plotted. I think this will be around 3 to 3.5 mb, which is a bit larger than the other options, but it's within the CRAN limit of 5 mb and keeps basically all of the data used in the case study accessible for the user to play with (this would be my other top choice)
3. If we're still interested in a separate repo for the manuscript that isn't a package, we could (at least for the case study data) use the reichlab/ensemble-comparison repo which has all of the same data used in the case study since that was a first draft of the work. It needs some cleaning up but could be a place to host the manuscript, or just the case study data objects

**Comment by nickreich** [comment]
I see your point. But shouldn't those toy example datasets be part of some hubUtils/hubData (which are imported by hubEnsembles)?

**Comment by elray1** [comment]
I feel like we're veering back to broader questions, so I added a comment on the other discussion [here](https://github.com/orgs/Infectious-Disease-Modeling-Hubs/discussions/6#discussioncomment-8389828).

I am more and more inclined to put this manuscript in a separate repo that's not hubEnsembles.  We could go back to what I think was an early vision for this:  `manuscript = hubEnsembles vignette + hubverse context + Li's flu analysis`.  We keep the part of the manuscript that really is specifically a hubEnsembles vignette as... a hubEnsembles vignette, in this repo.  We can update that vignette with content to match what we end up with in this manuscript.  The example data files that you want for the manuscript, as well as Li's flu analysis code and results, can also live in that repo.  We won't worry about adding extra stuff to hubEnsembles that distracts from its core mission of providing ensembling functionality.

**Comment by annakrystalli** [comment]
+1 from me on @elray1 comment. We can always link liberally throughout package docs to the paper and manuscript repository for additional context and materials to explore but I think we should strive to keep what's in the vignettes (with respect to content and organisation) focused on fast and efficient information retrieval about package features, functionality and deployment for someone trying to use the package. It would also solve the issue of adding a bunch of package dependencies to `Suggests` that are only manuscript related.

**Comment by lshandross** [comment]
I agree with @elray1 about a separate manuscript repo too. Removing the extra data files and unnecessary content for the manuscript for the vignette seems like the right way to go about this. I also like that it solves the `Suggests` issue. 

As far as implementing this solution, where do we want to house this repository, i.e. under which account, and what should we call it?

**Comment by elray1** [comment]
Nick and I discussed and we vote for putting this in Infectious-Disease-Modeling-Hubs.  It feels like a hubverse project more than a Reich lab project.

**Comment by lshandross** [comment]
Are there any best practices that @eahowerton and I should keep in mind when performing the split? Wondering if you have any tips @annakrystalli

**Comment by nickreich** [comment]
> In particular, I'm wondering if it makes more sense to address some of the issues about manuscript edits since they're connected to this repo. (This might not matter since we can link branches and pull requests in other repositories, but I just wanted to ask in case.)

You can transfer issues between repos, so it doesn't really matter from that perspective when the manuscript edits are made.

**Comment by annakrystalli** [comment]
@lshandross is preserving the git history of the `manuscript` contents what you are interested in?

**Comment by lshandross** [comment]
Yes, that's my primary interest/concern, but I'd also welcome any other insights you have from experience

**Comment by annakrystalli** [comment]
I believe you could use `git-subtree split`. Here's a [gist with some relevant code](https://gist.github.com/mvneves/23878af4789f6a5ab05079feb8097550) and [blog post](https://medium.com/@jeevansathisocial/extract-subdirectory-from-git-repository-without-losing-history-3de8aed359a4) with a bit more description around the process.

Note that by following the procedure you will effectively copy to contents of `manuscript` dir into the root of a new repo, which I believe is what you would want anyways.

**Comment by annakrystalli** [comment]
In terms of structuring the repo as a research compendium, I really like the structure/templates/tools provided by the [rrtools](https://github.com/benmarwick/rrtools) 📦. At the very least it has a nice README template for a repo accompanying a publication.
